File: Salary_Prediction.py
"""
Simple Linear Regression
Salary_Prediction
Author: Sola Gbenro
"""
# importing libraries
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

"""
    Import the dataset
"""
dataset = pd.read_csv('Salary_Data.csv')
# Matrix of independent variable
X = dataset.iloc[:, :-1].values
# Matrix of dependent variable
y = dataset.iloc[:, 1].values

"""
    Splitting the dataset
        
"""
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)


"""
    Fitting Simpe Linear Regression to the Training set
"""
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting the Test set results
y_pred = regressor.predict(X_test)

# visualizing the TRAINING set results
# x-axis years of experience, y-axis salary
plt.scatter(X_train, y_train, color = 'red')
# plotting against the training data NOT the test data
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()


# Visualizing the TEST set results
plt.scatter(X_test, y_test, color = 'red')
# same 'simple linear regression' line, using same regressor no need to change
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: Salary_Prediction.R
#   Simple Linear Regression
#   Salary_Prediction
#   Author: Sola Gbenro

#   Import dataset
#   indexing starts at 1
dataset = read.csv('Salary_Data.csv')
# dataset = dataset[, 2:3]

#   Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
#   wil return true or false depending on whether data was chosen for Training (true) or not (false) 
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#   Fitting Simple Linear Regression to the Training set
#   Salary ~(proporitional) to YearsExperience
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

#   Predicting the Test set results
#   vector of predictions
y_pred = predict(regressor, newdata = test_set)

#   Visualizing the Training set results
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Training set)') +
  xlab('Years of experience') +
  ylab('Salary')

#   Visualizing the Test set results
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Test set)') +
  xlab('Years of experience') +
  ylab('Salary')
